packages/rich-text/src/plugins/Heading/index.tsx

```diff
@@ -202,7 +202,11 @@ export function ToolbarHeadingButton(props: ToolbarHeadingButtonProps) {
       }
 
       toggleNodeType(editor, { activeType: type, inactiveType: type });
-      Slate.ReactEditor.focus(editor);
+
+      // TODO: Figure out why focus only works with timeout here.
+      setTimeout(() => {
+        Slate.ReactEditor.focus(editor);
```

Comment:
I think we can use `Transform.select` before. I asked the Plate team once about it, this might be helpful for you: https://slate-js.slack.com/archives/C013QHXSCG1/p1631022429081900